### DISCUSSION TITLE
fatal: unable to access 'https://gopkg.in/src-d/go-git.v4/': The requested URL returned error: 502

### DISCUSSION THREAD

**Discussion by yangjuexi** [community/community#56932] [Actions] [answered]
### Select Topic Area

Bug

### Body

There's something wrong with action.
My workflow can not fetch go dependencies and all get same error, here is the log 

```
# cd .; git clone https://gopkg.in/cas.v2 /home/runner/go/.cache/govendor/gopkg.in/cas.v2
Cloning into '/home/runner/go/.cache/govendor/gopkg.in/cas.v2'...
remote: Cannot obtain refs from GitHub: cannot talk to GitHub: Get https://github.com/go-cas/cas.git/info/refs?service=git-upload-pack: write tcp 10.131.9.188:60820->140.82.121.3:443: write: broken pipe
fatal: unable to access 'https://gopkg.in/cas.v2/': The requested URL returned error: 502

```

**Comment by curtismenmuir** [comment]
+1 seeing this issue today as well: 

```
remote: Cannot obtain refs from GitHub: cannot talk to GitHub: Get https://github.com/go-yaml/yaml.git/info/refs?service=git-upload-pack: write tcp 10.131.9.188:60820->140.82.121.3:443: write: broken pipe
fatal: unable to access 'https://gopkg.in/yaml.v2/': The requested URL returned error: 502
```

```
remote: Cannot obtain refs from GitHub: cannot talk to GitHub: Get https://github.com/go-playground/colors.git/info/refs?service=git-upload-pack: write tcp 10.131.9.188:60820->140.82.121.3:443: write: broken 
fatal: unable to access 'https://gopkg.in/go-playground/colors.v1/': The requested URL returned error: 502
```

**Comment by RachelZhao** [comment]
similar issue:
remote: Cannot obtain refs from GitHub: cannot talk to GitHub: Get https://github.com/go-yaml/yaml.git/info/refs?service=git-upload-pack: write tcp 10.131.9.188:60820->140.82.121.3:443: write: broken pipe

fatal: unable to access 'https://gopkg.in/yaml.v3/': The requested URL returned error: 502

**Reply by RachelZhao** [reply]
issue mitigated from my side

**Comment by yangjuexi** [comment]
So, do we have any other quicker ways to let Github know this issue? or, we can only wait ?

**Comment by skport** [comment]
Is there any way to quickly communicate this to the Github support team?

Or should I contact the owner of gopkg.in (https://labix.org/gopkg.in) ?

**Comment by curtismenmuir** [comment]
Here are some issue threads with the owner of gopkg.in:
- https://github.com/niemeyer/gopkg/issues/78
- https://github.com/niemeyer/gopkg/issues/79

also an issue with `go-yaml`: https://github.com/go-yaml/yaml/issues/887

**Comment by ureuzy** [comment, marked as answer]
I don't know why this problem occurred at this time, but at least going through Proxy solved it for me.
`GOPROXY=https://proxy.golang.org,direct`
This was default, but I had it set to direct only!

**Reply by omartrigui** [reply]
You don't have to specify `GOPROXY` env variable as it defaults to `https://proxy.golang.org,direct` (check https://go.dev/ref/mod#environment-variables)

**Comment by curtismenmuir** [comment]
Thank you so much @ureuzy 

I can confirm that setting `GOPROXY` Env Var to `https://proxy.golang.org,direct` resolved our issues too :) 

EG add `GOPROXY=https://proxy.golang.org,direct` to Jenkinsfile env vars

**Reply by harsh-rr** [reply]
our file already had this and still it is failing :(

**Reply by ureuzy** [reply]
@harsh-rr 
I don't know your environment, but from my local and on CI the proxy server appears to be alive.
Do you have same problem on your local machine? or it could be that the pkg you need is not in the proxy.
You could try using the following index to see if it is possible to get it via proxy.
https://index.golang.org/index

For other items to check, please read this one
https://proxy.golang.org/

**Comment by yangjuexi** [comment]
It seems all problems were resolved. i will close this issue. Thanks for your help
